#### marker alignment
setwd("~/OneDrive - University of Georgia/Bermudagrass/Comparative_Genomics")
# read sam file generated by Bowtie, aligning HapMap fasta to Sorghum
mySam <- readLines("Msinensis_output.sam")
mySam <- mySam[-(1:11240)] # get rid of header rows

# find which rows have successful alignments
aligned <- c(grep("Chr",mySam), grep("scaffold",mySam))
length(aligned)  # 32488
mySam[aligned[1:20]]
alignedC <- grep("Chr",mySam)
length(alignedC) # 29929

# convert into table
temp <- strsplit(mySam[alignedC], "\t")
postable <- data.frame(marker=sapply(temp,
                                     function(x) x[1]),
                       reference=sapply(temp, function(x) x[3]),
                       position=as.integer(sapply(temp, function(x) x[4])))
dim(postable)
postable[1:20,]
write.csv(postable, "Aligned 29929 tags on M. sinensis v7.0 genome.csv", row.names = FALSE)
